hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/SelectorOptimizer.java

```diff
@@ -163,6 +163,11 @@ public boolean remove(Object o) {
 
         @Override
         public boolean contains(Object o) {
+            for (int i = 0; i < activeKeys.size; i++) {
```

Comment:
Can you load the activeKeys.size into a constant and use that constant to iterate over the loop. Otherwise you are at risk of calling activeKeys.size() on every iteration of the loop unless the JIT is able to optimize it out.

And can you also store the activeKey.keys array into a local variable before the loop, to increase the chances that the JIT can optimize this code better.

Reply:
https://github.com/hazelcast/hazelcast/pull/24267/commits/395b9b77cf0c96d5f8954c42ab50a896dfda4960